storage/merge.go

```diff
@@ -751,13 +768,19 @@ func (c *compactChunkIterator) Next() bool {
 				heap.Push(&c.h, iter)
 			}
 		}
+		// Trust counter reset for first chunk
+		if len(c.h) > 0 {
+			c.prevIterator = c.h[0]
+		}
```

Comment:
I added this in as it was causing a few tests to fail like
https://github.com/prometheus/prometheus/blob/c037a3df844d15b5e38be7eaa28f842c5ee07ee2/storage/merge_test.go#L615-L626

I think it's okay to trust the counter reset for the first chunk. If that chunk overlaps with another chunk later on, it should be fixed later on. The first sample's counter reset doesn't matter if we're doing a rate() or increase() call.

Reply:
is this going to be the first sample in the block? because there's an argument by @beorn7 that it should be unknown reset. Blocks can be deleted by hand so the previous sample that caused the reset could be gone.